### PR TITLE
Go sdk upgrade and atlas-go-client update 

### PR DESCRIPTION
**Pull request by govardhanpagidi**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by ParthasarathyV — APPROVED** [review]
Go Lang -- Update from 1.13 to 1.19
Indirect dependencies based on the new version upgrade.

**Comment by gssbzn** [comment]
I don't see any tests running here, considering how big of a bump we are going with go version it would be nice to first stablish a base line in terms of test and make sure at least compilation works

File: cfn-resources/go.mod
Comment (andreaangiolillo) [inline cfn-resources/go.mod:13]: [q] Any reason why you are not using [v1.18](https://github.com/mongodb/go-client-mongodb-atlas/releases)?

File: cfn-resources/go.mod
Comment (andreaangiolillo) [inline cfn-resources/go.mod:28]: Could you add some tests? How do you know that you are not introducing breaking changes?

Comment (govardhanpagidi) [reply cfn-resources/go.mod:28]: After library upgrade I have tested PROJECT Resource, which has the Code change.

Test: CFN SUBMIT, updated the version in AWS (PrivateExtension) and Deployed Cloud Formation Template.

File: cfn-resources/go.mod
Comment (govardhanpagidi) [reply cfn-resources/go.mod]: Upgraded to 0.18.0 in the latest commit.

**Comment by Zuhairahmed** [comment]
What is purpose of cfn-resources/project/rpdk.log.1 log file here? it's 53K+ lines long. if we don't need can we remove? Otherwise LGTM pending APIx review.

**Comment by govardhanpagidi** [comment]
@Zuhairahmed , I've Removed the log file

**Review by Zuhairahmed — APPROVED** [review]
Thanks for updating @govardhanpagidi. LGTM!

File: cfn-resources/project/cmd/resource/resource.go
Comment (andreaangiolillo) [inline cfn-resources/project/cmd/resource/resource.go:33]: this does not seem related to the go upgrade. Is this supposed to be here?

File: cfn-resources/project/resource-role.yaml
Comment (andreaangiolillo) [inline cfn-resources/project/resource-role.yaml:24]: same as above. This is not related to the go upgrade, right?

File: cfn-resources/go.mod
Comment (andreaangiolillo) [reply cfn-resources/go.mod:28]: is it possible to add a github action that runs for every PR to test at least compilation? I am okay with a follow up PR

File: cfn-resources/project/cmd/resource/resource.go
Comment (govardhanpagidi) [reply cfn-resources/project/cmd/resource/resource.go:33]: this PR includes atlas-go-client library upgrade too, its bumped to v0.18.0

File: cfn-resources/go.mod
Comment (govardhanpagidi) [reply cfn-resources/go.mod:28]: @andreaangiolillo ,thanks for raising this. I ll raise another PR for the github action for builds, otherwise all the resources build is successful.

**Review by andreaangiolillo — APPROVED** [review]
LGTM. Thanks for all the changes

**Pull request merged**
